kernel/src/scan/log_replay.rs

```diff
@@ -241,22 +242,30 @@ impl LogReplayScanner {
 /// indicates whether the record batch is a log or checkpoint batch.
 pub fn scan_action_iter(
     engine: &dyn Engine,
-    action_iter: impl Iterator<Item = DeltaResult<(Box<dyn EngineData>, bool)>>,
+    action_iter: impl Iterator<Item = DeltaResult<(Box<dyn EngineData>, bool)>> + 'static,
     table_schema: &SchemaRef,
     predicate: Option<ExpressionRef>,
-) -> impl Iterator<Item = DeltaResult<ScanData>> {
+) -> Box<dyn Iterator<Item = DeltaResult<ScanData>>> {
```

Comment:
We should consider returning `DeltaResult<impl Iterator>`, so the error is immediately obvious instead of buried in the iterator itself. Plenty of examples of that elsewhere in the kernel.

Reply:
done!